cli/flox/src/commands/install.rs

```diff
@@ -107,11 +140,46 @@ impl Install {
                 })
             },
             Err(e @ EnvironmentSelectError::EnvNotFoundInCurrentDirectory) => {
-                bail!(formatdoc! {"
-                {e}
-
-                Create an environment with 'flox init' or install to an environment found elsewhere with 'flox install {} --dir <PATH>'",
-                self.packages.join(" ")})
+                let user_state_path = user_state_path(&flox);
+                let (lock, mut user_state) = lock_and_read_user_state_file(&user_state_path)?;
+                if user_state.confirmed_create_default_env.is_some() {
+                    bail!(formatdoc! {"
+                    {e}
+
+                    Create an environment with 'flox init' or install to an environment found elsewhere with 'flox install {} --dir <PATH>'",
+                    self.packages.join(" ")});
+                }
+                let msg = formatdoc! {"
+                    Packages must be installed into a Flox environment, which can be
+                    a user 'default' environment or attached to a directory.
+                "};
+                message::plain(msg);
+                let package_list = package_list_for_prompt(&packages_to_install)
+                    .context("must specify at least one package to install")?;
+                let (choice_idx, _) = Dialog {
+                    message: &format!(
+                        "Would you like to install {package_list} to the 'default' environment?"
+                    ),
+                    help_message: None,
+                    typed: Select {
+                        options: vec!["Yes", "No"],
+                    },
+                }
+                .raw_prompt()?;
```

Comment:
nonblocking: you could pass this in as a `prompt_function`, and then you could unit test:
1. Happy path creates a default environment. `PathEnvironment::init` does strike me as something that's brittle and could change
2. Doesn't prompt more than once (by making the prompt_function panic)